lib/ask/ecto_types/schedule.ex

```diff
@@ -143,37 +143,91 @@ defmodule Ask.Schedule do
     "Etc/UTC"
   end
 
-  def next_available_date_time(%Schedule{} = schedule, %DateTime{} = date_time \\ DateTime.utc_now) do
+  # Find the beggining of the first active window, going forward
+  def next_available_date_time(%Schedule{end_date: end_date, start_date: start_date} = schedule, %DateTime{} = from_date_time \\ SystemTime.time.now) do
+    from_date_time = select_from_date_time(from_date_time, start_date)
+    backward = false
+    limit = end_date
+    next_available_date = reversible_next_available_date_time(schedule, from_date_time, backward, limit)
+
+    if limit_exceed(next_available_date, backward, limit) do
+      raise ScheduleError, "next active window not found"
+    else
+      next_available_date
+    end
+  end
```

Comment:
This logic is repeated here and in `last_window_ends_up`. also, it seems leaked responsibility from `next_available_date`.

Why not instead of returning `erl_date` on `next_available_date` when `limit_exceed` we raise the `ScheduleError`? The error will bubble up and should have the same effect